src/test/java/core/basesyntax/service/impl/CsvTransactionParserImplTest.java

```diff
@@ -0,0 +1,101 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.CsvTransactionParser;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class CsvTransactionParserImplTest {
+    private static final List<String> INVALID_OPERATION_TRANSACTIONS =
+            List.of("type,fruit,quantity", "%$@#,banana,20");
+    private static final List<String> INVALID_INT_VALUE_TRANSACTIONS = List
+            .of("type,fruit,quantity", "b,banana,20word");
+    private static final List<String> INVALID_NUMBER_OF_VALUES_TRANSACTIONS =
+            List.of("type,fruit,quantity", "b,banana");
+    private static final String INVALID_NUMBER_OF_VALUES_EXCEPTION_MESSAGE =
+            "Incorrect number of data in row";
+    private static final String INVALID_OPERATION_EXCEPTION_MESSAGE =
+            "Such operation does not exist";
+    private static final String INVALID_INT_VALUE_EXCEPTION_MESSAGE =
+            "Wrong input type, int type expected";
+    private static CsvTransactionParser<List<FruitTransaction>, List<String>> csvTransactionParser;
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @BeforeClass
+    public static void beforeClass() {
+        csvTransactionParser = new CsvTransactionParserImpl();
+    }
+
+    @Test
+    public void csvParse_invalidOperationThrowException_notOk() {
+        expectedException.expect(RuntimeException.class);
+        csvTransactionParser.csvParse(INVALID_OPERATION_TRANSACTIONS);
+    }
+
+    @Test
+    public void csvParse_invalidIntegerValueThrowException_notOk() {
+        expectedException.expect(RuntimeException.class);
+        csvTransactionParser.csvParse(INVALID_INT_VALUE_TRANSACTIONS);
+    }
+
+    @Test
+    public void csvParse_invalidNumberOfDataInRowThrowException_notOk() {
+        expectedException.expect(RuntimeException.class);
+        csvTransactionParser.csvParse(INVALID_NUMBER_OF_VALUES_TRANSACTIONS);
+    }
+
+    @Test
+    public void readData_ExceptionMessageWithInvalidInvalidOperation_ok() {
+        expectedException.expectMessage(INVALID_OPERATION_EXCEPTION_MESSAGE);
+        csvTransactionParser.csvParse(INVALID_OPERATION_TRANSACTIONS);
+    }
+
+    @Test
+    public void readData_ExceptionMessageWithInvalidIntegerValue_ok() {
+        expectedException.expectMessage(INVALID_INT_VALUE_EXCEPTION_MESSAGE);
+        csvTransactionParser.csvParse(INVALID_INT_VALUE_TRANSACTIONS);
+    }
+
+    @Test
+    public void readData_ExceptionMessageWithInvalidNumberOfDataInRow_ok() {
```

Comment:
```suggestion
    public void readData_ExceptionMessageWithInvalidNumberOfDataInRow_notOk() {
```

Reply:
everything with exception in it must be marked "notOk"? because i was thinking about expected result mark, like message matches expected and this is "ok"

Reply:
then when should we write notOk?
notOk means that method doesn't do what it should, because of invalid input, for example

Reply:
got it